google/cloud/forseti/services/inventory/base/resources.py

```diff
@@ -2186,12 +2229,186 @@ class KmsCryptoKeyVersionIterator(resource_iter_class_factory(
     """The Resource iterator implementation for KMS CryptoKeyVersion."""
 
 
-class KubernetesClusterIterator(resource_iter_class_factory(
-        api_method_name='iter_container_clusters',
-        resource_name='kubernetes_cluster',
-        api_method_arg_key='projectNumber',
-        resource_validation_method_name='container_api_enabled')):
-    """The Resource iterator implementation for Kubernetes Cluster."""
+# class KubernetesClusterIterator(resource_iter_class_factory(
+#         api_method_name='iter_container_clusters',
+#         resource_name='kubernetes_cluster',
+#         api_method_arg_key='projectNumber',
+#         resource_validation_method_name='container_api_enabled')):
+#     """The Resource iterator implementation for Kubernetes Cluster."""
+
+
+class KubernetesClusterIterator(ResourceIterator):
+    """The Resource iterator implementation for KubernetesCluster"""
+
+    def iter(self):
+        """Resource iterator.
+
+        Yields:
+            Resource: KubernetesCluster created
+        """
+        gcp = self.client
+        try:
+            for data, metadata in gcp.iter_container_clusters(
+                    project_number=self.resource['id']):
+                yield FACTORIES['kubernetes_cluster'].create_new(
+                    data, metadata=metadata)
+        except ResourceNotSupported as e:
+            # API client doesn't support this resource, ignore.
+            LOGGER.debug(e)
+
+
+class KubernetesNodeIterator(ResourceIterator):
+    """The Resource iterator implementation for KubernetesCluster"""
+
+    def iter(self):
+        """Resource iterator.
+
+        Yields:
+            Resource: KubernetesCluster created
+        """
+        gcp = self.client
+        try:
+            for data, metadata in gcp.iter_k8s_nodes(
+                    project_number=self.resource['id'],
```

Comment:
It is project_id not project_number.

To pass project_id you need to use:
```
self.resource.parent()['id']
```

Reply:
Updated.